### PR TITLE
refactor: separate task types [DET-3395]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

required by the experiment list page.

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/pages/Dashboard.tsx [hunk 1]
```diff
@@ -17,8 +17,9 @@ import Users from 'contexts/Users';
 import useStorage from 'hooks/useStorage';
 import { ShirtSize } from 'themes';
 import {
-  Command, CommandState, RecentTask, ResourceType, RunState, TaskType,
+  Command, CommandState, CommandType, RecentTask, ResourceType,  RunState, TaskType,
```
Comment (hamidzr) [inline webui/react/src/pages/Dashboard.tsx:20]: extra space

File: webui/react/src/components/TaskActionDropdown.tsx [hunk 2]
```diff
@@ -56,32 +57,33 @@ const TaskActionDropdown: React.FC<Props> = ({ task }: Props) => {
             experimentId: parseInt(task.id),
```
Comment (hkang1) [inline webui/react/src/components/TaskActionDropdown.tsx:57]: chore: `setExperimentState` is defined as:
```
export const setExperimentState =
  async ({ state, ...rest }: PatchExperimentState): Promise<void> => {
    return patchExperiment({
      body: { state },
      ...rest,
    });
  };
```
It's missing `await` in front of `patchExperiment`.

File: webui/react/src/components/TaskActionDropdown.tsx [hunk 3]
```diff
@@ -56,32 +57,33 @@ const TaskActionDropdown: React.FC<Props> = ({ task }: Props) => {
...
             state: RunState.Active,
           });
-          await updateExperimentLocally(exp => ({ ...exp, state: RunState.Active }));
+          updateExperimentLocally(exp => ({ ...exp, state: RunState.Active }));
```
Comment (hkang1) [inline webui/react/src/components/TaskActionDropdown.tsx:60]: question: should `updateExperimentLocally` be defined as an `async` function and have `await` applied to `setExperiments`?

**Comment by hkang1** [comment]
@hamidzr ignore the comment about fixing the `await` issue, I'll have a [PR](https://github.com/determined-ai/determined/pull/739) that will address it.

File: webui/react/src/components/TaskActionDropdown.tsx (hunk 2 above)
Comment (hamidzr) [reply webui/react/src/components/TaskActionDropdown.tsx:57]: leaving it to https://github.com/determined-ai/determined/pull/739

**Pull request merged**